src/MetadataUtility/Audio/Wamd.cs

```diff
@@ -0,0 +1,203 @@
+        {
+            OffsetDateTime? date = null;
+
+            try
+            {
+                date = OffsetDateTimePattern.CreateWithInvariantCulture("yyyy'-'MM'-'dd' 'HH':'mm':'sso<m>").Parse(value).Value;
+            }
+            catch (UnparsableValueException)
+            {
+                return null;
+            }
+
+            return date;
+        }
+
+        /// <summary>
+        /// Parses a location's longitude and latitude.
+        /// Assigns the location data to a given wamd object.
+        /// </summary>
+        /// <param name="value">The location to parse.</param>
+        /// <param name="wamdData">The wamd object.</param>
+        public static void SetLocation(string value, Wamd wamdData)
+        {
+            string[] locationInfo = value.Split(",", StringSplitOptions.RemoveEmptyEntries);
+
+            double latitude = double.Parse(locationInfo[0]);
+            string latitudeDirection = locationInfo[1];
+
+            double longitude = double.Parse(locationInfo[2]);
+            string longitudeDirection = locationInfo[3];
+
+            wamdData.Latitude = latitudeDirection.Equals("N") ? latitude : latitude * -1;
+            wamdData.Longitude = latitudeDirection.Equals("E") ? longitude : longitude * -1;
+        }
+
+        /// <summary>
+        /// Extracts metadata from a wamd chunk.
+        /// </summary>
+        /// <param name="wamdSpan">The wamd chunk.</param>
+        public static Wamd ExtractMetadata(ReadOnlySpan<byte> wamdSpan)
+        {
+            Wamd wamdData = new Wamd();
+
+            // Links metadata set functions to their corresponding subchunk ID
+            Dictionary<int, Action<string>> setters = new Dictionary<int, Action<string>>
+            {
+                { 1, value => wamdData.Name = value },
+                { 2, value => wamdData.SerialNumber = value },
+                { 3, value => wamdData.Firmware = value },
+                { 5, value => wamdData.StartDate = DateParser(value) },
+                { 18, value => wamdData.MicrophoneType = value.Split(",") },
+                { 19, value => wamdData.MicrophoneSensitivity = value.Split(",") },
+                { 20, value => SetLocation(value, wamdData) },
+                { 21, value => wamdData.Temperature = value },
+            };
+
+            int wamdOffset = 0;
+            ushort subChunkId;
+            uint length;
+            string value;
+
+            // Parse each piece of metadata in the wamd chunk
+            while (wamdOffset < wamdSpan.Length)
+            {
+                subChunkId = BinaryPrimitives.ReadUInt16LittleEndian(wamdSpan[wamdOffset..]);
+                wamdOffset += 2;
+
+                length = BinaryPrimitives.ReadUInt32LittleEndian(wamdSpan[wamdOffset..]);
+                wamdOffset += 4;
+
+                int start = wamdOffset;
+                int end = wamdOffset + (int)length;
+
+                value = Encoding.ASCII.GetString(wamdSpan[start..end]);
```

Comment:
HMMM some fields could be UTF-8 https://github.com/QutEcoacoustics/emu/blob/b132d18c1f188c461446c490cc6ece40fadfee7d/docs/WildlifeAcousticsMetadataHeader/wa_meta.h#L97-L99

Not sure what to do about that...

Reply:
It looks like none of the current definitions use that provision, but maybe leave a todo to future editors that utf-8 may need to be selectively used for decoding the value.